### PR TITLE
Apply changes from P3187R1

### PR DESCRIPTION
**Pull request by lewissbaker**
Removes the following facilities as specified in P3187R1:
* `ensure_started()`
* `execute()`
* `start_detached()`

Also removes the following facility as it does not make sense any more:
* `execute_may_block_caller`

Note: `ensure_started()` algorithm had wording shared with `split()`
The common wording has now been simplified to now only describe `split()`.

### CODE REVIEW

**Review by inbal2l — APPROVED** [review]

**Review by ericniebler — CHANGES_REQUESTED** [review]
for thoroughness we should also remove mentions of `start_detached`, `execute`, and `ensure_started` from other parts of the paper.

File: execution.bs [hunk 1]
```diff
@@ -8634,47 +8570,41 @@ request but has no recommendataions at present.</span>
               the `ref_count` to `0` then synchronizes with
               the call to <code><i>dec-ref</i>()</code> that decrements `ref_count` to `0`.
 
-8. For each type `split_t` and `ensure_started_t`, there is a different,
-    associated exposition-only implementation tag type, <i>`split-impl-tag`</i>
-    and <i>`ensure-started-impl-tag`</i>, respectively. Let
-    <i>`shared-impl-tag`</i> be the associated implementation tag type of
-    <i>`shared-cpo`</i>. Given an expression `sndr`, the expression
-    <code><i>shared-cpo</i>.transform_sender(sndr)</code> is equivalent to:
```
Comment (ericniebler) [inline execution.bs:8642]: this and the below is wrong. we need a _`split-impl-tag`_, and the _`impls-for`_ below needs to be specialized for it.

File: execution.bs [hunk 2]
```diff
@@ -8464,46 +8421,33 @@ request but has no recommendataions at present.</span>
...
         void <i>complete</i>(Tag, Args&&... args) noexcept { <i>// exposition only</i>
+          using tuple_t = <i>decayed-tuple</i>&lt;Tag, Args...>;
           try {
-            using tuple_t = <i>decayed-tuple</i>&lt;Tag, Args...>;
             sh_state->result.template emplace&lt;tuple_t>(Tag(), std::forward&lt;Args>(args)...);
           } catch (...) {
-            using tuple_t = tuple&lt;set_error_t, exception_ptr>;
-            sh_state->result.template emplace&lt;tuple_t>(set_error, current_exception());
+            using err_tuple_t = tuple&lt;set_error_t, exception_ptr>;
+            sh_state->result.template emplace&lt;err_tuple_t>(set_error, current_exception());
```
Comment (ericniebler) [inline execution.bs:8450]:
```suggestion
            using tuple_t = tuple&lt;set_error_t, exception_ptr>;
            sh_state->result.template emplace&lt;tuple_t>(set_error, current_exception());
```

**Comment by lewissbaker** [comment]
> for thoroughness we should also remove mentions of `start_detached`, `execute`, and `ensure_started` from other parts of the paper.

I wasn't sure whether to leave them there but just add notes that they are no longer in the design or just remove them wholesale.

File: execution.bs [hunk 3]
```diff
@@ -8707,37 +8637,18 @@ request but has no recommendataions at present.</span>
...
-                calls `set_stopped(std::move(rcvr))` and returns.
-
-            3.  Otherwise, atomically does the following:
+            2.  Then atomically does the following:
 
-                  - Inserts `&state` into `state.sh_state->waiting_states`, and
+                  - Reads the value, `c`, of `state.sh_state->completed`.
 
-                  - Reads the value of `state.sh_state->completed`.
+                  - If `c` is `false`, inserts `&state` into `state.sh_state->waiting_states`.
```
[the remark this answers is not in the document]
Comment (lewissbaker) [reply execution.bs]: The code immediately below the 'atomic' block which handles the `c` is `true` case.

[the remark this answers is not in the document]
Comment (ericniebler) [reply execution.bs]: It amounts to the same thing, no?

[the remark this answers is not in the document]
Comment (ericniebler) [reply execution.bs]: no, it's not! i see the point of your change now. thanks for spotting this bug.

**Review by ericniebler — APPROVED** [review]

**Pull request merged**